src/data-import.ts

```
@@ -0,0 +1,331 @@
+ */
+export function isFileSystemAccessSupported(): boolean {
+  return typeof window !== 'undefined' && 'showOpenFilePicker' in window;
+}
+
+/**
+ * Open file picker using File System Access API or fallback
+ */
+export async function openFilePicker(): Promise<File | null> {
+  if (isFileSystemAccessSupported()) {
+    try {
+      const [fileHandle] = await (window as any).showOpenFilePicker({
+        types: [
+          {
+            description: 'Montana Inventory Files',
+            accept: {
+              'application/json': ['.json', '.txt'],
+              'text/plain': ['.json', '.txt'],
+            },
+          },
+        ],
+        excludeAcceptAllOption: true,
+        multiple: false,
+      });
+      
+      return await fileHandle.getFile();
+    } catch (error) {
+      if ((error as Error).name === 'AbortError') {
+        // User cancelled - not an error
+        return null;
+      }
+      throw error;
+    }
+  } else {
+    // Fallback to traditional file input
+    return new Promise((resolve) => {
+      const input = document.createElement('input');
+      input.type = 'file';
+      input.accept = '.json,.txt,application/json,text/plain';
+      
+      input.onchange = (event) => {
+        const file = (event.target as HTMLInputElement).files?.[0];
+        resolve(file || null);
+      };
+      
+      input.oncancel = () => resolve(null);
+      
+      input.click();
+    });
+  }
+}
+
+/**
+ * Parse and validate import file content
+ */
+export async function parseImportFile(file: File): Promise<ImportData> {
+  if (!file) {
+    throw new Error('Nessun file selezionato');
+  }
+
+  // Check file size (max 10MB for safety)
+  if (file.size > 10 * 1024 * 1024) {
+    throw new Error('File troppo grande. Dimensione massima: 10MB');
+  }
+
+  let content: string;
+  try {
+    content = await file.text();
+  } catch (error) {
+    throw new Error('Errore nella lettura del file');
+  }
```

Don't catch the error in this case, let the error bubble up.